packages/api/src/beacon/routes/beacon/block.ts

```diff
@@ -444,7 +444,16 @@ export function getDefinitions(config: ChainForkConfig): RouteDefinitions<Endpoi
           };
         },
         parseReqJson: ({body, headers}) => {
-          const fork = toForkName(fromHeaders(headers, MetaHeader.Version));
+          let fork: ForkName;
+          // As per spec, version header is optional for JSON requests
+          const versionHeader = fromHeaders(headers, MetaHeader.Version, false);
+          if (versionHeader !== undefined) {
+            fork = toForkName(versionHeader);
+          } else {
+            // Determine fork from slot in JSON payload
+            fork = config.getForkName((body as SignedBlindedBeaconBlock).message.slot);
```

Comment:
this is why I mean it's a hack btw, we kinda just yolo it here and try to access `body.message.slot` but this will result in "Cannot access property x of undefined" if data is not formatted correctyl, while if you get version from header we can properly apply validation before we try to access any property of body.